### PR TITLE
[WIP] Dependency version outdate highlighter

### PR DESCRIPTION
**Pull request by ylil93**
For #53 
todo:
- tests
- documentation
- run linter

### CODE REVIEW

**Comment by liyanhui1228** [comment]
@ylil93 Please sign the CLA, thanks!

File: compatibility_lib/compatibility_lib/compatibility_store.py [hunk 1]
```diff
@@ -408,3 +408,40 @@ def save_compatibility_statuses(
...
+    def get_dependency_info(self, package_name):
+        """Returns dependency info for an indicated internal package.
+
+        Args:
+            package_name: The package to lookup for.
+
+        Returns:
+            A mapping between the dependency names and the info (dict).
+        """
+
```
[the remark this answers is not in the document]
Comment (ylil93) [reply compatibility_lib/compatibility_lib/compatibility_store.py]: Done

File: compatibility_lib/compatibility_lib/compatibility_checker.py [hunk 2]
```diff
@@ -52,13 +52,15 @@ def retrying_check(self, args):
         python_version = args[1]
         return self.check(packages, python_version)
 
-    def get_self_compatibility(self, python_version):
+    def get_self_compatibility(self, python_version, packages=None):
+        if packages is None:
+            packages = configs.PKG_LIST
         """Get the self compatibility data for each package."""
```
[the remark this answers is not in the document]
Comment (ylil93) [reply compatibility_lib/compatibility_lib/compatibility_checker.py]: Done

File: compatibility_lib/compatibility_lib/compatibility_checker.py [hunk 3]
```diff
@@ -74,3 +76,11 @@ def get_pairwise_compatibility(self, python_version):
 
             for result in zip(pkg_set_results):
                 yield result
+
+    def get_dependency_info(self, package_name, python_version):
```
[the remark this answers is not in the document]
Comment (ylil93) [reply compatibility_lib/compatibility_lib/compatibility_checker.py]: deleted `get_dependency_info ` and refactored code to use `get_self_compatibility`

File: compatibility_lib/compatibility_lib/compatibility_store.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (liyanhui1228) [reply compatibility_lib/compatibility_lib/compatibility_store.py]: Thanks! Please make sure to push the commits before replying to comments, which makes it clearer which comments are resolved.

File: compatibility_lib/dependency_highlighter.py [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+from datetime import datetime
+import enum
+
+DATETIME_FORMAT = "%Y-%m-%d"
+DEFAULT_GRACE_PERIOD_IN_DAYS = 183
+MAJOR_GRACE_PERIOD_IN_DAYS = 30
+ALLOWED_MINOR_DIFF = 3
+
+class Priority(enum.Enum):
+    NONE = "UP_TO_DATE"
```
[the remark this answers is not in the document]
Comment (ylil93) [reply compatibility_lib/dependency_highlighter.py]: How about `NOT_SET`? I considered using `UP_TO_DATE`, but I thought that `Priority` should indicate a level of urgency and `UP_TO_DATE` sounds more like a status.

[the remark this answers is not in the document]
Comment (liyanhui1228) [reply compatibility_lib/dependency_highlighter.py]: That sounds good.

File: compatibility_lib/dependency_highlighter.py [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+
+from compatibility_lib import compatibility_checker
+from compatibility_lib import compatibility_store
+from compatibility_lib import configs
+
+from datetime import datetime
+import enum
+
+DATETIME_FORMAT = "%Y-%m-%d"
+DEFAULT_GRACE_PERIOD_IN_DAYS = 183
```
[the remark this answers is not in the document]
Comment (ylil93) [reply compatibility_lib/dependency_highlighter.py]: > Is this default grace period for everything else than major version grace period?

Yes - I was using the issue requirements to set that parameter
`Mark “High Priority” if a new version has been available for more than 6 months.` - https://github.com/GoogleCloudPlatform/cloud-opensource-python/issues/53

Comment added

File: compatibility_lib/compatibility_lib/compatibility_store.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ylil93) [reply compatibility_lib/compatibility_lib/compatibility_store.py]: Oh ok sorry about that! I'll hide the spammy comments.

**Pull request closed without merging**